packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -822,4 +825,17 @@ export class GraphQLTransform {
       index++;
     }
   }
+
+// Checks if DataStore is enabled and CPK Feature Flag is true. Throws a warning otherwise.
+public validateCPKFeatureFlag = (context: TransformerContext ) => {
+  const isDataStoreEnabled = context.isProjectUsingDataStore();
+  if (isDataStoreEnabled && !isCPKFeatureEnabled(context) && context.isProjectUsingCPK()) {
+    printer.info(
+      `
+⚠️  WARNING: Your schema has a custom primary key but the Feature Flag "${cpkFeatureFlagName}" is disabled. Check the value in your "amplify/cli.json" file, change it to "true" and re-run
```

Comment:
I would clarify what we mean by `re-run` here. For example:

```suggestion
⚠️  WARNING: Your schema has a custom primary key but the Feature Flag "${cpkFeatureFlagName}" is disabled. Check the value in your "amplify/cli.json" file, change it to `true`, and execute `amplify codegen models` with the CLI
```

@renebrandel any thoughts on the wording, here? Thanks!

Reply:
Another CX:
For some customers using the old Amplify projects -->
manually switching the FF to true can also be a breaking change and might lose some data. Should we also warn them regarding this? 

thank you @AaronZyLee for catching this perception

Reply:
@AaronZyLee - can you elaborate more on this?

Reply:
@manaswi223 - thoughts on the suggestion I made here (i.e. asking the customer to re-run `amplify codegen models`?